components/application-authenticators/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticator.java

```diff
@@ -168,14 +169,17 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             "Defaulting to Name Identifier.");
                 }
             }
+            idpSubject = (String) request.getSession().getAttribute("username");
             if (subject == null) {
-                subject = (String) request.getSession().getAttribute("username");
+                subject=idpSubject;
```

Comment:
Formatting issues.. need a white space before and after the '='. Please check other places too.